Convert photorec to use RawDiskPartition

dfjxs opened the pull request:
Issue #78

aarontp commented:
As you mentioned elsewhere, the previous changes are showing up here, so it's a bit hard to review.   I'm guessing you created this branch from the previous feature branch and not directly from master?   Maybe you could try syncing your master branch, then creating a new branch from there and then cherry-picking the actual new commits?  Maybe github is calculating the diffs based on the commits rather than the actual code diff here, and since those commits were squashed when merged into upstream master, it still thinks they are "new"?

dfjxs commented:
Sorry about that, rebased the branch. Should be good now.

dfjxs commented on turbinia/workers/photorec.py:
```diff
@@ -44,16 +40,23 @@ def run(self, evidence, result):
     # Add the output path to the evidence so we can automatically save it
     # later.
-    output_evidence.local_path = output_file_path
+    output_evidence.local_path = ''.join((output_file_path, '.1'))
```
Photorec appends .1 to whatever output folder specified.
I guess this is to support multiple runs with an incrementing number on the output folder.

dfjxs commented:
@aarontp I've moved the extraction of volume extents into the partition enumeration task and stored the values in the evidence item.

I also changed the logic a little to support images that are taken at the volume level (i.e. no partition table).

aarontp reviewed (commented):
Just a few small comments.

aarontp commented on turbinia/processors/mount_local.py:
```diff
@@ -48,17 +50,18 @@ def PreprocessLosetup(source_path):
   if not os.path.exists(source_path):
     raise TurbiniaException(
-        'Cannot process non-existing source_path {0!s}'.format(source_path))
+        'Cannot mount non-existing source_path {0!s}'.format(source_path))
```
minor nit (sorry I missed this last time): We're not actually mounting it in this preprocessor.  Maybe 'Cannot create loopback for ..."?

aarontp commented on turbinia/workers/photorec.py:
```diff
@@ -44,23 +42,32 @@ def run(self, evidence, result):
           shell=True)
-      output_evidence.compress()
+      if os.path.exists(output_evidence.local_path):
+        output_evidence.compress()
```
We should probably add a message here indicating that there was no output, and we might want to update the status message below to indicate that as well.

aarontp reviewed (commented):
Just a couple minor things.

dfjxs commented:
Fixed

aarontp reviewed (approved):
LGTM!
